[Test failure] [Daily] [dev12] Legacy build PreFAST errors

Branched from issue #528 

http://dotnet-ci.cloudapp.net/job/Microsoft_ChakraCore/job/daily_dev12_x64_release/51/consoleFull#1026559514ac1851af-571d-472f-b664-6657b078a5dd

```
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.cpp(254): warning C6102: Using 'bcBuffer' from failed function call at line '179'. [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.cpp(338): warning C6102: Using 'bcBuffer' from failed function call at line '319'. [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\chakrartinterface.cpp(28): warning C6102: Using 'modulename' from failed function call at line '27'. [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(52): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'ftell'. See line 51 for an earlier location where this can occur [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(51): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fseek'.  [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(64): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fread'. See line 51 for an earlier location where this can occur [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(65): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fclose'. See line 51 for an earlier location where this can occur [d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\ch.vcxproj]
20:38:56 [daily_dev12_x---5f770fb3] $ powershell.exe -NonInteractive -ExecutionPolicy ByPass "& 'C:\Users\DOTNET~1\AppData\Local\Temp\2\hudson7084396863305529629.ps1'"
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(52): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'ftell'. See line 51 for an earlier location where this can occur
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(51): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fseek'. 
20:38:56 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(64): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fread'. See line 51 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\bin\ch\helpers.cpp(65): warning C6387: 'file' could be '0':  this does not adhere to the specification for the function 'fclose'. See line 51 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\common\memory\arenaallocator.cpp(1196): warning C28182: Dereferencing NULL pointer. 'cache' contains the same NULL value as 'curWeakRef' did. See line 1196 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\common\memory\arenaallocator.cpp(1276): warning C28182: Dereferencing NULL pointer. 'cache' contains the same NULL value as 'curWeakRef' did. See line 1276 for an earlier location where this can occur
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\common\memory\recyclerweakreference.h(233): warning C28182: Dereferencing NULL pointer. 'buckets' contains the same NULL value as 'current' did. 
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\jsrt\jsrt.cpp(863): warning C28196: The requirement that '*_Param_(2)!=0' is not satisfied. (The expression does not evaluate to true.)
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\runtime\bytecode\bytecodegenerator.cpp(3733): warning C6011: Dereferencing NULL pointer 'invertedStmt'. 
20:38:57 d:\j\workspace\daily_dev12_x---5f770fb3\lib\runtime\language\dynamicprofileinfo.cpp(586): warning C6011: Dereferencing NULL pointer 'callSiteInfo[callSiteId].u.polymorphicCallSiteInfo'. See line 578 for an earlier location where this can occur
```

--- a/netci.groovy
+++ b/netci.groovy
@@ -28,7 +28,7 @@ def dailyRegex = 'dailies'
 // HELPER CLOSURES
 // ---------------
 
-def CreateBuildTasks = { machine, configTag, buildExtra, testExtra, excludeConfigIf, nonDefaultTaskSetup ->
+def CreateBuildTasks = { machine, configTag, buildExtra, testExtra, runCodeAnalysis, excludeConfigIf, nonDefaultTaskSetup ->
     [true, false].each { isPR ->
         ['x86', 'x64', 'arm'].each { buildArch ->
             ['debug', 'test', 'release'].each { buildType ->
@@ -43,7 +43,7 @@ def CreateBuildTasks = { machine, configTag, buildExtra, testExtra, excludeConfi
                 def jobName = Utilities.getFullJobName(project, config, isPR)
 
                 def testableConfig = buildType in ['debug', 'test'] && buildArch != 'arm'
-                def analysisConfig = buildType in ['release']
+                def analysisConfig = buildType in ['release'] && runCodeAnalysis
 
                 def buildScript = "call .\\jenkins\\buildone.cmd ${buildArch} ${buildType}"
                 buildScript += buildExtra ?: ''
@@ -144,14 +144,14 @@ def CreateStyleCheckTasks = { taskString, taskName, checkName ->
 // INNER LOOP TASKS
 // ----------------
 
-CreateBuildTasks('Windows_NT', null, null, null, null, null)
+CreateBuildTasks('Windows_NT', null, null, null, true, null, null)
 
 // -----------------
 // DAILY BUILD TASKS
 // -----------------
 
 // build and test on Windows 7 with VS 2013 (Dev12/MsBuild12)
-CreateBuildTasks('Windows 7', 'daily_dev12', ' msbuild12', ' -win7',
+CreateBuildTasks('Windows 7', 'daily_dev12', ' msbuild12', ' -win7', false,
     /* excludeConfigIf */ { isPR, buildArch, buildType -> (buildArch == 'arm') },
     /* nonDefaultTaskSetup */ { newJob, isPR, config ->
         DailyBuildTaskSetup(newJob, isPR,
@@ -159,14 +159,16 @@ CreateBuildTasks('Windows 7', 'daily_dev12', ' msbuild12', ' -win7',
             '(dev12|legacy)\\s+tests')})
 
 // build and test on the usual configuration (VS 2015) with -includeSlow
-CreateBuildTasks('Windows_NT', 'daily_slow', null, ' -includeSlow', null,
+CreateBuildTasks('Windows_NT', 'daily_slow', null, ' -includeSlow', false,
+    /* excludeConfigIf */ null,
     /* nonDefaultTaskSetup */ { newJob, isPR, config ->
         DailyBuildTaskSetup(newJob, isPR,
             "Windows ${config}",
             'slow\\s+tests')})
 
 // build and test on the usual configuration (VS 2015) with JIT disabled
-CreateBuildTasks('Windows_NT', 'daily_disablejit', ' "/p:BuildJIT=false"', ' -disablejit', null,
+CreateBuildTasks('Windows_NT', 'daily_disablejit', ' "/p:BuildJIT=false"', ' -disablejit', true,
+    /* excludeConfigIf */ null,
     /* nonDefaultTaskSetup */ { newJob, isPR, config ->
         DailyBuildTaskSetup(newJob, isPR,
             "Windows ${config}",

